### PR TITLE
Add support for new NCBI ranks that were recently introduced

### PR DESCRIPTION
**Pull request by pverscha**
A variety of different NCBI ranks were recently introduced, breaking the database build process. This PR updates the backend and fixes the database build process.

```
      1 subkingdom
      1 superphylum
      3 subcohort
      4 superkingdom
      5 cohort
      5 pathogroup
      5 subvariety
      6 superclass
      9 series
     11 kingdom
     12 morph
     17 biotype
     18 infraclass
     20 genotype
     21 subsection
     26 parvorder
     33 subphylum
     52 superorder
     71 serogroup
    124 speciessubgroup
    129 infraorder
    160 subclass
    264 phylum
    329 speciesgroup
    369 suborder
    402 section
    423 class
    506 subtribe
    544 forma
    739 formaspecialis
    861 superfamily
    885 clade
   1188 serotype
   1309 isolate
   1597 subgenus
   1634 order
   2195 tribe
   3036 subfamily
   8395 varietas
   9389 family
  24857 subspecies
  44244 strain
  95352 genus
 221432 norank
1853272 species
```

All newly added ranks that are associated with less than 100 taxa are ignored and set to "no rank". More specifically: "biotype", "serogroup", "morph", "genotype", "subvariety", "pathogroup".

Following ranks were not ignored and added: "clade", "strain", "isolate", "serotype" and "forma specialis". The order of these ranks however should be further reviewed, as this is not published anywhere. I tried do derive the order from the given taxon dump files, but it seems like some of the ranks cannot be ordered hierarchically:

```
In taxonomy (the science of naming and classifying biology), the word “clade” isn’t used only to divide a species. It can be used at a level from kingdom down to genus as well to identify a division below an official place where the tree of life divides.
```

### CODE REVIEW

**Comment by ninewise** [comment]
If the order of the ranks is inconsistent within the NCBI taxonomy, I think we'll have to leave those inconsistent ranks out (or set them to 'no rank' only where they are inconsistent).

**Comment by pverscha** [comment]
Since "strain" is the only rank that's defined clearly and that fits into our hierarchy in a non-ambiguous way, I decided to only include this rank and ignore the others until these are clarified by the NCBI.

The ranks "infraclass" and "parvorder" are also added to the list of ignored ranks since they occur very very seldom.

File: frequency.py [hunk 1]
```diff
@@ -2,12 +2,12 @@
 from collections import defaultdict
 
 order = [
-	["subspecies", "varietas", "forma"],
+	["subspecies", "strain", "varietas", "forma"],
```
Comment (pverscha) [inline frequency.py:5]: @ninewise Is this the correct place to add "strain" in this file? What is this file used for?

**Review by bmesuere — APPROVED** [review]

File: frequency.py (hunk 1 above)
Comment (ninewise) [reply frequency.py:5]: This script is used to show the frequency of ranks in an aggregated index. I used it to evaluate other aggregation methods on the backend for the sequences table. Eventually I replaced the LCA* aggregation for the metagenomics backend with a 95% LCA* hybrid approach. We haven't evaluated the use to this for the metaproteomics backend, yet.

And yes, this is the correct place to add 'strain'.

**Review by ninewise — CHANGES_REQUESTED** [review]
The RANKS constant and the `29` in `src/main/java/org/unipept/tools/LineagesSequencesTaxons2LCAs.java` may need to be changed, too? Unless we're not including 'strain' in the calculations of the LCA*.

**Comment by pverscha** [comment]
@ninewise I've updated the LCA calculations.

**Review by ninewise — APPROVED** [review]

**Review by ninewise — APPROVED** [review]

**Comment by pdawyndt** [comment]
Strain definitely is not a rank we can use for aggregation, I think. This indicates an individual, e.g. a single clone of a bacterium.

**Pull request merged**